modules/data/config.py

```diff
@@ -0,0 +1,216 @@
+"""Read the bot configuration from the settings.yaml and the reactions.yaml files"""
+import os
+import re
+from typing import Any, Literal, Union
+import yaml
+
+SettingsKeys = Literal["debug", "meme", "test", "token", "bot_tag"]
+SettingsDebugKeys = Literal["local_log", "reset_on_load"]
+SettingsMemeKeys = Literal["channel_group_id", "channel_id", "channel_tag", "comments", "group_id", "n_votes",
+                           "remove_after_h", "tag", "report", "report_wait_mins", "replace_anonymous_comments",
+                           "delete_anonymous_comments",]
+SettingsTestKeys = Literal[Literal["api_hash", "api_id", "session", "bot_tag", "token"], SettingsMemeKeys]
+SettingsKeyType = Union[SettingsKeys, tuple[Literal["debug", "meme", "test"], Literal[SettingsMemeKeys, SettingsDebugKeys,
+                                                                                      SettingsTestKeys]]]
+
+ReactionKeys = Literal["reactions", "rows"]
+ReactionKeyType = Union[ReactionKeys, tuple[Literal["reactions"], int]]
+
+
+class Config():
```

Comment:
I am not a fan of singleton, what's the point of using this pattern in Python in respect to having a set of functions?

Reply:
I guess that is just a personal preference. I feel like this way is more elegant, since it is easier to hide complexity from the end user and it is more straight forward to store variables in the object, instead of the class